src/test/objectstore/test_bluefs.cc

```diff
@@ -642,6 +642,32 @@ TEST(BlueFS, test_compaction_sync) {
     writes_done = true;
     join_all(sync_threads);
     fs.compact_log();
+
+    {
+      string dir = "dir.after_compact_test";
+      ASSERT_EQ(0, fs.mkdir(dir));
+      string file = "file.after_compact_test";
+      BlueFS::FileWriter *h;
+      ASSERT_EQ(0, fs.open_for_write(dir, file, &h, false));
+      ASSERT_NE(nullptr, h);
+      auto sg = make_scope_guard([&fs, h] { fs.close_writer(h); });
+      h->append("some data", 9);
```

Comment:
Data string should be defined as const variable before calling h->append() and we can use strlen() instead of the magic number 9